File: app/src/main/java/com/example/myapplication4/Clases/Gasto.java
package com.example.myapplication4.Clases;
import java.util.ArrayList;
import java.math.BigDecimal;
import java.time.LocalDate;

public class Gasto extends Transaccion{
    private int cantCuotas;
    private BigDecimal interes;
    private ArrayList<Cuota> cuotas;

    public Gasto(int id, String desc, BigDecimal monto, LocalDate fecha, int cantCuotas, BigDecimal interes, Categoria categoria){
        super(id, desc, monto, fecha, cantCuotas, interes, categoria);

        cuotas = new ArrayList<Cuota>();
        LocalDate fechaPagar = LocalDate.now();
        for(int i=0;i<cantCuotas;i++){
            Cuota cuota = new Cuota(i+1,fechaPagar.plusMonths(i));
        }
    }
}

File: app/src/main/java/com/example/myapplication4/clases/Gasto.kt
package com.example.myapplication4.clases
import java.time.LocalDate

// Clase hija: Gasto
data class Gasto(
// Propiedades heredadas de Transaccion
    val idGasto: Int,
    val descGasto: String,
    val montoGasto: Float,
    val fechaGasto: LocalDate,
    val categoriaGasto: Categoria,
//Propiedades propias
    val cantCuotas: Int,
    val interes: Float,
    val cuotas: ArrayList<Cuota> = arrayListOf()
) : Transaccion(idGasto, descGasto, montoGasto, fechaGasto, categoriaGasto) {
    init {
        require(cantCuotas > 0) { "El número de cuotas debe ser mayor que 0" }
        val fechaPagar = LocalDate.now()
        for (i in 0 until cantCuotas) {
            val cuota = Cuota(i + 1, fechaPagar.plusMonths(i.toLong()))
            cuotas.add(cuota)
        }
    }
}
